src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -99,7 +110,12 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                if (!Task.Run(() => context.CurrentResult = innerCommand.Execute(context)).Wait(_timeout))
+                var testExecution = RunTestWithTimeoutAsync(context);
+                if (_debugger.IsAttached)
+                {
+                    AwaitCompletionWithoutExceptionWrapping(testExecution);
+                }
```

Comment:
This wasn't the most straightforward for me to see because the pieces are spread out, but functionally it's solid.

Reply:
When revisiting the code, I figured I have issues understanding it either. I have changed the code a bit, I'm still not entirely happy with it but I hope it's easier to understand now. I hope you don't mind that I change it.